nautobot/extras/health_checks.py

```diff
@@ -65,10 +66,32 @@ def check_status(self):
                 obj.title = "newtest"
                 obj.save()
                 obj.delete()
-        except IntegrityError:
-            raise ServiceReturnedUnexpectedResult("Integrity Error")
-        except DatabaseError:
-            raise ServiceUnavailable("Database error")
+        except IntegrityError as e:
+            self.add_error(ServiceReturnedUnexpectedResult("Integrity Error"), e)
+        except DatabaseError as e:
+            self.add_error(ServiceUnavailable("Database error"), e)
+
+
+class MigrationsBackend(NautobotHealthCheckBackend):
+    """Check whether all migrations have been applied."""
+
+    metric = Gauge(
+        "health_check_migrations_info",
+        "State of migrations backend",
+        multiprocess_mode=NautobotHealthCheckBackend.MULTIPROCESS_MODE,
+    )
+
+    def check_status(self):
+        db_alias = getattr(settings, "HEALTHCHECK_MIGRATIONS_DB", DEFAULT_DB_ALIAS)
+        try:
+            executor = MigrationExecutor(connections[db_alias])
+            plan = executor.migration_plan(executor.loader.graph.leaf_nodes())
+            if plan:
+                self.add_error(ServiceUnavailable("There are migrations not yet applied"))
```

Comment:
Does this impact development of new migrations? I'm typically rolling a migration back, starting the web server to add some rest data, then testing my in-flight migration. This won't prevent the server from starting with unapplied migrations right? Biggest impact might be beat and celery workers not starting up?

Reply:
I believe you're correct. This shouldn't prevent the server from starting, but it will prevent the docker health check from reporting as healthy until you apply your migration. I think this could be worked around easily enough with a docker-compose.override.yml but not 100% certain.

Reply:
As currently implemented, this doesn't actually check the health of the active nautobot server process. If that check was implemented this migrations check wouldn't be required. Are you planning to implement that as a future enhancement? That was one of our items from the design discussion:

> Doing a nautobot-server healthcheck management command - should check both that the “real” server process is running and also that things like DB connectivity are working

Reply:
That may be desirable as a separate command - for the Docker case we don't need to check the server process health since the container will die if the server process exits, and I think we don't in general have a way to detect "health" of the process other than "is it running", as "running and busy processing HTTP requests" is mostly indistinguishable externally from "running and stuck in an infinite loop somewhere"